### ISSUE TITLE
Dwarfs do not mine out all the ore in the background when building buildings.

### ISSUE THREAD

**Issue by Barry-2707** [WarpzoneStudios/Hammerting#29]
**Describe the bug**
When you place a building in the world all the tiles that are supposed to be mined out are marked but they are not always  mined out before the building is complete. This wastes valueble resources.

**To Reproduce**
1. Place a random building in the world (preferably over ore tiles more than 4 tiles from the ground)
2. Watch them mine out some tiles but not all
3. Wait until all resources are deliverd and watch them start to build it.
4. Most of the time this building is finished before all tiles are mined out and  the remaining ones just disappear in thin air.
5. Observe that you have lost valueble resources.

**Expected behavior**
Dwarfs should mine out all necessary tiles before building can start/is completed so that you do not waste resources.

**Screenshots**
![image](https://user-images.githubusercontent.com/78851197/107856719-5fcec400-6e2a-11eb-9c1c-c64d7e0e13dd.png)

**Comment by dustingraham** [comment]
I find this a personal decision. If I'm in a hurry, I place the building, and let the dwarves build it ASAP, sometimes getting the ore, sometimes not.

If I have a little more time, and want to ensure all of the ore is mined, I place the building, then cancel the building, and build scaffolds as needed in order to get to all of the minerals since some may be higher than the dwarves can reach.

Seems like blocking the building if there are any ore patches could lead to similar problems. "Building never completes!" this is a tricky one. I don't see it as a bug from my perspective.

To add on to this, I can see a couple other problems.

Depending on how it is patched, if all resources are instant-mined when the building is completed, it could be used as a hack to quickly mine a ton of resources by placing a low cost building over a patch of minerals.

On the other hand, if there isn't a way to skip mining the ore, it seems like it would slow down the game and be frustrating. I have 600+ copper sometimes, I'd rather not waste the time building scaffolds to reach one small patch blocking my building 8 tiles up.

Not sure what the solution would be for this -- honestly I'm happy with my current workflow. Place the building, cancel it, wait for it to be mined out, then place the building again once the area is clear.

**Comment by Barry-2707** [comment]
I totally agree with you that it is a personal dicision and I also do not know what the best way of solving this would be. But please understand that this is not solely my idea I simply added it here after people pointed it out on Discord.
